### PR TITLE
Added backup functionality of database

### PR DESCRIPTION
**Pull request by esunar**
removed backup related code from zaza_utils

### CODE REVIEW

**Review by rgildein — APPROVED** [review]
LGTM

File: cou/steps/backup.py [hunk 1]
```diff
@@ -15,8 +15,72 @@
...
+    logging.info("mysqldump mysql-innodb-cluster DBs ...")
+    action = model.run_action_on_leader(mysql_app, "mysqldump")
+    remote_file = action.data["results"]["mysqldump-file"]
+    basedir = action.data["parameters"]["basedir"]
+
+    logging.info("Set permissions to read mysql-innodb-cluster:%s ...", basedir)
+    model.run_on_leader(mysql_app, f"chmod o+rx {basedir}")
+
+    local_file = os.path.abspath(os.path.basename(remote_file))
+    logging.info("SCP from  mysql-innodb-cluster:%s to %s ...", remote_file, local_file)
```
Comment (gabrielcocenza) [inline cou/steps/backup.py:44]: nit: double space at `"SCP from  mysql-innodb-cluster`

File: tests/unit/steps/test_steps_backup.py [hunk 2]
```diff
@@ -1,9 +1,84 @@
...
+                        "placement-mysql-router",
+                    ],
+                },
+            }
+        }
+        app = get_database_app()
+        assert app == "mysql"
+
+
+def test_get_database_app_negative():
```
Comment (gabrielcocenza) [inline tests/unit/steps/test_steps_backup.py:50]: Can you add some comments why percona it's not selected in this case?
Thanks

**Review by gabrielcocenza — APPROVED** [review]
LGTM.

Just added some comments that aren't blockers.

**Pull request merged**